pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,131 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	ApiVip            string `json:"apiVip,omitempty"`
+	ApiVipDnsName     string `json:"apiVipDnsName,omitempty"`
+	IngressVip        string `json:"ingresVip,omitempty"`
```

Comment:
These three items above look to be very metal specific to me. Installer has these within the BareMetal platform object. If we keep them here as is, it implies that an Agent based install platform is metal specific. I've heard reports that Agent based installs could be done on any cloud though, is this a feature we wish to leave the door open for in the API? I'm not clear why anyone would do that when there's a cloud API in play. It also waters down the installer's purpose, so I'm really unsure if we should focus on it here.

If we do want to keep that door open, we have an unfortunate platform within a platform requirement... How could we model that sanely?

Suggestion is to leave these where they are and consider Agent based install to be metal specific, but open to alternatives.

Reply:
My team has flated the idea of this being an "InstallStrategy" instead of a "Platform", leaving "Platform" open to alternate implementations beyond bare metal. Not totally sure how to structure this, we already have a baremetal platform which has some similarities to what Agent metal installs want, but not much, and most of it would be incompatible. This would like mean an "Agent" install strategy and a new "AgentBareMetal" platform.

Reply:
From call today this looks ok, will make the change asap.